### ISSUE TITLE
how to set ignore_label in tensorflow?

### ISSUE THREAD

**Issue by zuokai** [ppwwyyxx/tensorpack#596]
when i set the label=-1, there is an error: Received a label value of -1 which is outside the valid range of [0, 8)

**Comment by ppwwyyxx** [comment]
I don't know what do you mean by "ignore". Write your math down and convert it to tensorflow.

You might be interested in `tf.gather`,  `tf.boolean_mask`.

**Comment by zuokai** [comment]
if I set the label = -1 of a picture， that mean this picture does not calculate the loss

**Comment by ppwwyyxx** [comment]
Again, look at `tf.gather`, `tf.boolean_mask`.

The documentation of tensorflow is very clear that `tf.nn.sparse_softmax_cross_entropy_with_logits` does not support -1.

Not a tensorpack question. Closing.
